services/priceService.ts

```diff
@@ -62,23 +61,121 @@ export async function upsertCurrentPricesForNetworkId (responseData: IResponseDa
   })
 }
 
-export async function syncCurrentPrices (): Promise<boolean> {
+export async function upsertCurrentPricesForNetworkId (responseData: IResponseData, networkId: number): Promise<void> {
+  await upsertPricesForNetworkId(responseData, networkId, 0)
+}
+
+function validateTicker (ticker: string): void {
+  if (!Object.values(TICKERS).includes(ticker)) {
+    throw new Error(RESPONSE_MESSAGES.INVALID_TICKER_400.message)
+  }
+}
+
+function validatePriceAPIUrlAndToken (): void {
   if (appInfo.priceAPIURL === '') {
     throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_URL_400.message)
   }
-  const todayString = moment().format(PRICE_API_DATE_FORMAT)
+  if (appInfo.priceAPIToken === '') {
+    throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_TOKEN_400.message)
+  }
+}
+
+function getPriceByDayURL (day: moment.Moment, ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
```

Comment:
Nice, moved them both to validators. 

About the rename, I don't agree with it, because it only validates if the ticker is a ticker we support, i.e. if it is in our const list of tickers. If that ticker has a network or quote related to it or not is a separate issue.

Reply:
I don't understand, those tickers are network tickers, not quote tickers. E.g: XEC is a network ticker; in the future could be a quote ticker also (meaning e.g. we could show the dashboard value of all paybuttons in XEC). This const list of tickers are the tickers of the networks we support. `validateTicker` is going to fail for `USD`, which is a quote ticker we support.

Reply:
Now I understood better what you meant, makes sense, done